src/newFunctionality.js

```diff
@@ -0,0 +1,38 @@
+import Storage from '../storage.js';
+
+let arr = [];
+if (Storage.isEmpty()) {
+  Storage.set(arr);
+}
+
+class ToDo {
+  constructor(description) {
+    this.description = description;
+    this.checked = false;
+    this.index = null;
+  }
+}
+
+const newToDo = (value) => {
+  const todo = new ToDo(value);
+  arr = Storage.get();
+  arr.push(todo);
+  Storage.set(arr);
+};
+
+const todoDelete = (btnId, btnNode) => {
+  btnNode.remove();
+  let storageGet = Storage.get();
+  storageGet = storageGet.filter((elem) => elem.index !== Number(btnId));
+  Storage.set(storageGet);
+};
+
+const edit = (edElement) => {
+  edElement.setAttribute('contenteditable', 'true');
+};
```

Comment:
- [ ] This was a requirement from the [previous review](https://github.com/isaka-lumato/to-do-list/pull/3/files#r693048797). I can see you've tried to add a method to edit a task but the logic in it still cannot edit a task, it's only setting `contenteditable` to true. You need to add logic to actually persist edit changes to localStorage.

Take a look at the below suggested change and see how we can use `input` and `focusout` event listeners to achieve this.

```suggestion
const edit = (edElement) => {
  const arr = Storage.get();
  let newDescription = '';
  edElement.addEventListener('input', () => {
    newDescription = edElement.textContent;
  });
  edElement.addEventListener('focusout', () => {
    if (newDescription) {
      const task = arr[edElement.id - 1];
      task.description = newDescription;
      Storage.set(arr);
      newDescription = '';
      window.location.reload();
    }
  });
};
```